### PR TITLE
Support EventBridge events in lambdas/es/indexer, test event shape

### PR DESCRIPTION
**Pull request by akarve**
# Description
* Support EventBridge events in lambdas/es/indexer
* Use JSON Schemas to validate inbound events
* Fix a critical bug where delete events were not sent to ElasticSearch

# TODO
- [x] Documentation (add notes on EventBridge set up)
- [x] [Changelog](CHANGELOG.md) entry

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -13,7 +13,9 @@
 ## CLI
 
 ## Catalog, Lambdas
+* [Added] Support for EventBridge S3 events to es/indexer([#1987](https://github.com/quiltdata/quilt/pull/1987))
```
Comment (nl0) [inline docs/CHANGELOG.md:16]:
```suggestion
* [Added] Support for EventBridge S3 events to es/indexer ([#1987](https://github.com/quiltdata/quilt/pull/1987))
```

File: lambdas/es/indexer/document_queue.py [hunk 2]
```diff
@@ -193,10 +211,59 @@ def send_all(self):
...
+                                    # use match (not term) because some of these fields are analyzed
+                                    {"match": {"handle": handle}},
+                                    {"match": {"pointer_file": pointer_file}},
+                                    {"match": {"delete_marker": False}},
+                                ]
+                            }
+                        }
+                    },
+                    # we delete synchronously, so don't let it linger too long
+                    timeout='20s'
```
Comment (kevinemoore) [inline lambdas/es/indexer/document_queue.py:245]: What happens if this times out?

**Review by kevinemoore — APPROVED** [review]

File: lambdas/search/index.py [hunk 3]
```diff
@@ -235,6 +235,8 @@ def post_process(result: dict, action: str) -> dict:
         # and all other extensions are single-level
         corrected.extend(gz)
         result["aggregations"]["exts"]["buckets"] = corrected
+    elif action == "packages":
```
Comment (nl0) [inline lambdas/search/index.py:238]: is this change intentional?

File: lambdas/es/indexer/requirements.txt [hunk 4]
```diff
@@ -6,24 +6,25 @@ certifi==2019.6.16
 chardet==3.0.4
 decorator==4.4.0
 docutils==0.15.2
-elasticsearch==6.8.1
+elasticsearch==6.3.1
```
Comment (sir-sigurd) [inline lambdas/es/indexer/requirements.txt:9]: Please leave a comment why it's needed.

File: lambdas/es/indexer/document_queue.py (hunk 2 above)
Comment (akarve) [reply lambdas/es/indexer/document_queue.py:245]: We don't do the delete. I could do this async.

File: lambdas/es/indexer/index.py [hunk 5]
```diff
@@ -445,6 +577,42 @@ def make_s3_client():
...
+            format_checker=draft7_format_checker,
+        )
+    except ValidationError as error:
+        logger_.error("Invalid event format: %s\n%s", error, event)
+        return None
+    # be a good citizen and don't modify params
+    shaped = event.copy()
+    shaped["eventName"] = map_event_name(event)
+
+    return shaped
```
Comment (sir-sigurd) [inline lambdas/es/indexer/index.py:613]:
```suggestion
    return {
        **event,
        'eventName': map_event_name(event),
    }
```

File: lambdas/search/index.py (hunk 3 above)
Comment (akarve) [reply lambdas/search/index.py:238]: No, I'll remove it.

File: lambdas/es/indexer/document_queue.py [hunk 6]
```diff
@@ -102,24 +107,34 @@ def append(
...
+            index_name += PACKAGE_INDEX_SUFFIX
         if not index_name:
             raise ValueError(f"Can't infer index name; bucket={bucket}, doc_type={doc_type}")
+        # core properties for all document types;
+        # see https://elasticsearch-py.readthedocs.io/en/6.3.1/helpers.html
         body = {
             "_index": index_name,
+            "_op_type": _op_type,  # determines if action is upsert (index) or delete
             "_type": "_doc",
+            # TODO nest fields under "document" and maybe use _type:{package, object}
```
[the remark this answers is not in the document]
Comment (akarve) [reply lambdas/es/indexer/document_queue.py:124]: We don't use it and it doesn't throw an error but I'm reluctant to remove it as I believe the API barfed last time I tried it. Note that our AWS-side ES version is pinned to 6.7 so, as long as that holds, we're OK.
